common/primitives/src/messages.rs

```diff
@@ -103,12 +106,42 @@ impl<BlockNumber, T> BlockPaginationResponse<BlockNumber, T> {
 			next_index: None,
 		}
 	}
+
+	/// Checks if we are at the end of the pagination
+	/// if we are, update the response with the correct next information
+	pub fn check_end_condition_and_set_next_pagination(
```

Comment:
This was moved from an independent function, to one here on the `BlockPaginationResponse` struct as that is what it mutates.